### PR TITLE
feat(private-sql.activation): private SQL activation for db over 512MB

### PR DESCRIPTION
**Pull request by lucas-chaigne**
WEB-13675

Signed-off-by: Lucas Chaigne <lucas.chaigne@ovhcloud.com>

- [ ] AGORA CATALOG TO BE UPDATED BEFORE MERGE IN PROD

| Question         | Answer
| ---------------- | ---
| Branch?          | develop
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | WEB-13675
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

## Description

Customer can activate for free a private SQL 512M when he orders a perf hosting. We want to add new products : customer will be able to activate private SQL over 512M at a lower price when he has a perf hosting. This PR goal is to update manager private-sql-activation page to reflect this change.

### CODE REVIEW

File: packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.controller.js [hunk 1]
```diff
@@ -1,146 +1,75 @@
...
-      .finally(() => {
-        this.loading.checkout = false;
-      });
+  getOptionsValues() {
+    return this.privateSqlOptions.map((e) => {
+      return {
+        display: this.getPlanCodeDisplay(e.planCode),
+        planCode: e.planCode,
+      };
+    });
```
Comment (anooparveti) [inline packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.controller.js:69]:
```suggestion
    return this.privateSqlOptions.map((e) => {({
        display: this.getPlanCodeDisplay(e.planCode),
        planCode: e.planCode,
      }));
```

**Review by anooparveti — DISMISSED** [review]

**Review by anooparveti — DISMISSED** [review]

File: packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.html [hunk 2]
```diff
@@ -40,34 +37,9 @@
                 data-items="$ctrl.versions"
                 data-match="label"
                 data-searchable
+                required
```
Comment (y4nnL) [inline packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.html:40]:
```suggestion
                data-required
```

File: packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.routing.js [hunk 3]
```diff
@@ -31,12 +31,36 @@ export default /* @ngInject */ ($stateProvider) => {
...
           PrivateDatabase.getOrderableDatabaseVersions('classic'),
-        services: /* @ngInject */ (OvhApiHostingWeb) =>
-          OvhApiHostingWeb.v6()
-            .query()
-            .$promise.then((services) =>
-              orderBy(services, (serviceName) => toLower(serviceName)),
+        datacenter: /* @ngInject */ (serviceName, getDatacenter) =>
+          getDatacenter(serviceName),
+        getDatacenter: /* @ngInject */ (Hosting) => async (serviceName) => {
+          const { datacenter } = await Hosting.getHosting(serviceName);
```
Comment (y4nnL) [inline packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.routing.js:37]: **question**: are you sure this is AngularJS friendly ? I mean, no issues with the digest cycle or the dirty checking ?

File: packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.html [hunk 4]
```diff
@@ -1,32 +1,29 @@
...
+>
     <oui-step-form
-        data-header="{{:: 'privatesql_activation_step_1' | translate }}"
-        data-valid="$ctrl.hosting && $ctrl.option && $ctrl.version && !$ctrl.error.hosting"
-        data-prevent-next="true"
-        data-loading="$ctrl.loading.hosting"
-        data-editable="!$ctrl.loading.checkout"
-        data-on-submit="$ctrl.fetchCheckoutInformations()"
+        header="{{:: 'privatesql_activation_step_1' | translate }}"
+        editable="$ctrl.options.isEditable"
```
[the remark this answers is not in the document]
Comment (lucas-chaigne) [reply packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.html]: Hello, thanks for the review, it's updated

File: packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.html [hunk 5]
```diff
@@ -1,32 +1,29 @@
...
 </oui-back-button>
-<oui-stepper>
+
+<ovh-manager-product-offers
+    pricing-type="$ctrl.productOffers.pricingType"
+    workflow-options=":: $ctrl.productOffers.workflowOptions"
+    workflow-type=":: $ctrl.productOffers.workflowType"
+    on-success=":: $ctrl.onSuccess(checkout)"
+    on-error=":: $ctrl.onError(error)"
+    send-current-state=":: $ctrl.getOrderState(state)"
```
[the remark this answers is not in the document]
Comment (lucas-chaigne) [reply packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.html]: Hello, thanks for the review, it's updated

File: packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.html [hunk 6]
```diff
@@ -1,32 +1,29 @@
...
-            <p
-                data-ng-bind="$ctrl.error.hosting.data.message"
-                data-ng-if="$ctrl.error.hosting.data.message"
-            ></p>
-        </oui-message>
         <oui-field
-            data-size="xl"
-            data-label="{{:: 'privatesql_activation_price_select' | translate }}"
+            label="{{:: 'privatesql_activation_price_select' | translate }}"
+            size="xl"
```
[the remark this answers is not in the document]
Comment (lucas-chaigne) [reply packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.html]: Hello, thanks for the review, it's updated

File: packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.html [hunk 7]
```diff
@@ -1,32 +1,29 @@
...
             <oui-select
                 name="optionSelect"
-                data-model="$ctrl.option"
-                data-items="$ctrl.options"
-                data-disabled="$ctrl.loading.hosting"
-                data-match="label"
+                model="$ctrl.options.value"
+                items="$ctrl.options.values"
+                match="display"
+                required
```
[the remark this answers is not in the document]
Comment (lucas-chaigne) [reply packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.html]: Hello, thanks for the review, it's updated

File: packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.html (hunk 2 above)
Comment (lucas-chaigne) [reply packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.html]: Hello, thanks for the review, it's updated

File: packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.routing.js (hunk 3 above)
Comment (lucas-chaigne) [reply packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.routing.js:37]: Hello, I took this cope snippet from https://github.com/ovh/manager/blob/master/packages/manager/apps/web/client/app/hosting/database/order/private/hosting-database-order-private.routing.js

**Review by anooparveti — DISMISSED** [review]

**Review by y4nnL — DISMISSED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**